### ISSUE TITLE
False negative for `deprecated-module` when a `__import__` method is used instead of `import` sentence.

### ISSUE THREAD

**Issue by moylop260** [pylint-dev/pylint#10453]
### Bug description

```python
# pylint: disable=invalid-name

__import__("mypackage")
```

### Configuration

```ini
[Imports]
deprecated-modules=mypackage
```

### Command used

```shell
pylint -d all -e deprecated-module
```

### Pylint output

```python
# Empty output
```

### Expected behavior

`Deprecated module 'mypackage' (deprecated-module)`

This style is used from popular auto-code snippets:
 - https://github.com/honza/vim-snippets/blob/f0a3184/snippets/python.snippets#L209

### Pylint version

```shell
pylint 4.0.0-dev0
astroid 4.0.0a0
Python 3.12.11 (main, Jun  3 2025, 15:41:47) [Clang 17.0.0 (clang-1700.0.13.3)]
```

### OS / Environment

MACOSX

### Additional dependencies

```python

```
